lib/pure/os.nim

```diff
@@ -3217,6 +3184,41 @@ proc getFileInfo*(path: string, followSymlink = true): FileInfo {.noWeirdTarget.
         raiseOSError(osLastError(), path)
     rawToFormalFileInfo(rawInfo, path, result)
 
+proc sameFileContent*(path1, path2: string; checkSize = false; bufferSize = 8192.Positive): bool {.
+    rtl, extern: "nos$1", tags: [ReadIOEffect], noWeirdTarget.} =
+  ## Returns true if both pathname arguments refer to files with identical
+  ## binary content.
+  ##
+  ## If `checkSize` is `true` then checks the file sizes *before reading the files*,
+  ## if the files have different file sizes they can not have the same contents,
+  ## `checkSize = true` may be faster, specially for very big files.
+  ##
+  ## See also:
+  ## * `sameFile proc <#sameFile,string,string>`_
+  var a, b: File
+  var mustRead = true
+  var bufA = alloc(bufferSize)
+  var bufB = alloc(bufferSize)
+  try:  # readBuffer or open may or may not raise IOError.
+    if not(open(a, path1)) or not(open(b, path2)): mustRead = false
+    if checkSize and getFileInfo(a).size != getFileInfo(b).size: mustRead = false
```

Comment:
i think this should be moved up 1 line, no need to `open` if `getFileInfo(x).size` differs

Reply:
`getFileInfo` is taking `File`.